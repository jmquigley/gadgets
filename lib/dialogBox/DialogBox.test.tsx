'use strict';

import {shallow} from 'enzyme';
import * as React from 'react';
import {DialogBox, DialogBoxType, getDefaultDialogBoxProps} from './index';

test('Test retrieval of DialogBox props object', () => {
	const props = getDefaultDialogBoxProps();

	expect(props).toBeDefined();
	expect(props).toMatchSnapshot();
});

test('Test creation of a default DialogBox control', () => {
	const ctl = shallow(<DialogBox className="test-class" />);

	expect(ctl).toBeDefined();
	expect(ctl).toMatchSnapshot();
});

test('Test the creation of an error DialogBox', () => {
	const ctl = shallow(
		<DialogBox
			dialogType={DialogBoxType.error}
			message="Sample error message"
			show
		/>
	);

	expect(ctl).toBeDefined();
	expect(ctl).toMatchSnapshot();
});

test('Test the creation of an warning DialogBox', () => {
	const ctl = shallow(
		<DialogBox
			dialogType={DialogBoxType.warning}
			message="Sample warning message"
			show
		/>
	);

	expect(ctl).toBeDefined();
	expect(ctl).toMatchSnapshot();
});

test('Test the creation of an success DialogBox', () => {
	const ctl = shallow(
		<DialogBox
			dialogType={DialogBoxType.success}
			message="Sample success message"
			show
		/>
	);

	expect(ctl).toBeDefined();
	expect(ctl).toMatchSnapshot();
});

test('Test the creation of an info DialogBox', () => {
	const ctl = shallow(
		<DialogBox
			dialogType={DialogBoxType.info}
			message="Sample info message"
			show
		/>
	);

	expect(ctl).toBeDefined();
	expect(ctl).toMatchSnapshot();
});

test('Test the creation of an custom DialogBox', () => {
	const ctl = shallow(
		<DialogBox
			dialogType={DialogBoxType.custom}
			iconName="car"
			message="Sample info message"
			show
			style={{
				color: 'magenta'
			}}
		/>
	);

	expect(ctl).toBeDefined();
	expect(ctl).toMatchSnapshot();
});

test('Test pressing the "yes" button on the default DialogBox', () => {
	const selection = jest.fn();
	const ctl = shallow(
		<DialogBox
			message="Testing click handler for yes"
			onSelection={selection}
			show
		/>
	);

	expect(ctl).toBeDefined();
	expect(ctl).toMatchSnapshot();
	expect(ctl.state('showModal')).toBe(true);
});
