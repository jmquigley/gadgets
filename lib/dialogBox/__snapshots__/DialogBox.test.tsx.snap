// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test creation of a default DialogBox control 1`] = `
<Wrapper
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className="test-class"
  color={null}
  contentEditable={false}
  controlled={true}
  dialogType="info"
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  iconName="bomb"
  id={null}
  left={null}
  location="none"
  message=""
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="DialogBox"
  onClose={[Function]}
  onError={[Function]}
  onOpen={[Function]}
  onSelection={[Function]}
  padding={null}
  reset={false}
  ripple={false}
  selected={false}
  show={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
>
  <Modal
    ariaHideApp={false}
    bodyOpenClassName="ReactModal__Body--open"
    closeTimeoutMS={0}
    contentLabel="DialogBox"
    isOpen={false}
    onAfterOpen={[Function]}
    onRequestClose={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={false}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {
          "borderRadius": "0",
          "bottom": "auto",
          "left": "50%",
          "marginRight": "-50%",
          "padding": "15px",
          "right": "auto",
          "top": "40%",
          "transform": "translate(-50%, -50%)",
        },
      }
    }
  >
    <DialogBox__DialogBoxView
      className="ui-dialogbox test-class"
    >
      <DialogBox__DialogBoxIconView>
        <DialogBox__StyledIcon
          iconName="info-circle"
          sizing="xxlarge"
          style={
            Object {
              "color": "#5bc0de",
            }
          }
        />
      </DialogBox__DialogBoxIconView>
      <DialogBox__DialogBoxMessageContainerView>
        <DialogBox__DialogBoxMessageView
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p></p>",
            }
          }
        />
        <DialogBox__ButtonBar>
          <DialogBox__StyledButtonText
            justify={2}
            noicon={true}
            onClick={[Function]}
            sizing="normal"
            text="Yes"
          />
          <DialogBox__Spacer />
          <DialogBox__StyledButtonText
            justify={2}
            noicon={true}
            onClick={[Function]}
            sizing="normal"
            text="No"
          />
        </DialogBox__ButtonBar>
      </DialogBox__DialogBoxMessageContainerView>
    </DialogBox__DialogBoxView>
  </Modal>
</Wrapper>
`;

exports[`Test pressing the "yes" button on the default DialogBox 1`] = `
<Wrapper
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  dialogType="info"
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  iconName="bomb"
  id={null}
  left={null}
  location="none"
  message="Testing click handler for yes"
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="DialogBox"
  onClose={[Function]}
  onError={[Function]}
  onOpen={[Function]}
  onSelection={[MockFunction]}
  padding={null}
  reset={false}
  ripple={false}
  selected={false}
  show={true}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
>
  <Modal
    ariaHideApp={false}
    bodyOpenClassName="ReactModal__Body--open"
    closeTimeoutMS={0}
    contentLabel="DialogBox"
    isOpen={true}
    onAfterOpen={[Function]}
    onRequestClose={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={false}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {
          "borderRadius": "0",
          "bottom": "auto",
          "left": "50%",
          "marginRight": "-50%",
          "padding": "15px",
          "right": "auto",
          "top": "40%",
          "transform": "translate(-50%, -50%)",
        },
      }
    }
  >
    <DialogBox__DialogBoxView
      className="ui-dialogbox"
    >
      <DialogBox__DialogBoxIconView>
        <DialogBox__StyledIcon
          iconName="info-circle"
          sizing="xxlarge"
          style={
            Object {
              "color": "#5bc0de",
            }
          }
        />
      </DialogBox__DialogBoxIconView>
      <DialogBox__DialogBoxMessageContainerView>
        <DialogBox__DialogBoxMessageView
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>Testing click handler for yes</p>",
            }
          }
        />
        <DialogBox__ButtonBar>
          <DialogBox__StyledButtonText
            justify={2}
            noicon={true}
            onClick={[Function]}
            sizing="normal"
            text="Yes"
          />
          <DialogBox__Spacer />
          <DialogBox__StyledButtonText
            justify={2}
            noicon={true}
            onClick={[Function]}
            sizing="normal"
            text="No"
          />
        </DialogBox__ButtonBar>
      </DialogBox__DialogBoxMessageContainerView>
    </DialogBox__DialogBoxView>
  </Modal>
</Wrapper>
`;

exports[`Test retrieval of DialogBox props object 1`] = `
Object {
  "backgroundColor": null,
  "borderColor": null,
  "bottom": null,
  "children": null,
  "className": null,
  "color": null,
  "contentEditable": false,
  "controlled": true,
  "dialogType": "info",
  "direction": "right",
  "disabled": false,
  "err": null,
  "errorMessage": null,
  "focus": false,
  "height": null,
  "iconName": "bomb",
  "id": null,
  "left": null,
  "location": "none",
  "message": "",
  "minHeight": null,
  "minWidth": null,
  "noedit": false,
  "nohover": false,
  "notheme": false,
  "notooltip": false,
  "obj": "DialogBox",
  "onClose": [Function],
  "onOpen": [Function],
  "onSelection": [Function],
  "padding": null,
  "ripple": false,
  "selected": false,
  "show": false,
  "sizing": "normal",
  "style": Object {},
  "testing": true,
  "theme": null,
  "tooltip": null,
  "top": null,
  "visible": true,
  "width": null,
  "xcss": null,
}
`;

exports[`Test retrieval of DialogBox props object 2`] = `
Object {
  "children": null,
  "showModal": false,
  "sizing": "normal",
  "style": Object {},
}
`;

exports[`Test the creation of an custom DialogBox 1`] = `
<Wrapper
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  dialogType="custom"
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  iconName="car"
  id={null}
  left={null}
  location="none"
  message="Sample info message"
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="DialogBox"
  onClose={[Function]}
  onError={[Function]}
  onOpen={[Function]}
  onSelection={[Function]}
  padding={null}
  reset={false}
  ripple={false}
  selected={false}
  show={true}
  sizing="normal"
  style={
    Object {
      "color": "magenta",
    }
  }
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
>
  <Modal
    ariaHideApp={false}
    bodyOpenClassName="ReactModal__Body--open"
    closeTimeoutMS={0}
    contentLabel="DialogBox"
    isOpen={true}
    onAfterOpen={[Function]}
    onRequestClose={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={false}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {
          "borderRadius": "0",
          "bottom": "auto",
          "left": "50%",
          "marginRight": "-50%",
          "padding": "15px",
          "right": "auto",
          "top": "40%",
          "transform": "translate(-50%, -50%)",
        },
      }
    }
  >
    <DialogBox__DialogBoxView
      className="ui-dialogbox"
    >
      <DialogBox__DialogBoxIconView>
        <DialogBox__StyledIcon
          iconName="car"
          sizing="xxlarge"
          style={
            Object {
              "color": "magenta",
            }
          }
        />
      </DialogBox__DialogBoxIconView>
      <DialogBox__DialogBoxMessageContainerView>
        <DialogBox__DialogBoxMessageView
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>Sample info message</p>",
            }
          }
        />
        <DialogBox__ButtonBar>
          <DialogBox__StyledButtonText
            justify={2}
            noicon={true}
            onClick={[Function]}
            sizing="normal"
            text="Yes"
          />
          <DialogBox__Spacer />
          <DialogBox__StyledButtonText
            justify={2}
            noicon={true}
            onClick={[Function]}
            sizing="normal"
            text="No"
          />
        </DialogBox__ButtonBar>
      </DialogBox__DialogBoxMessageContainerView>
    </DialogBox__DialogBoxView>
  </Modal>
</Wrapper>
`;

exports[`Test the creation of an error DialogBox 1`] = `
<Wrapper
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  dialogType="error"
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  iconName="bomb"
  id={null}
  left={null}
  location="none"
  message="Sample error message"
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="DialogBox"
  onClose={[Function]}
  onError={[Function]}
  onOpen={[Function]}
  onSelection={[Function]}
  padding={null}
  reset={false}
  ripple={false}
  selected={false}
  show={true}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
>
  <Modal
    ariaHideApp={false}
    bodyOpenClassName="ReactModal__Body--open"
    closeTimeoutMS={0}
    contentLabel="DialogBox"
    isOpen={true}
    onAfterOpen={[Function]}
    onRequestClose={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={false}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {
          "borderRadius": "0",
          "bottom": "auto",
          "left": "50%",
          "marginRight": "-50%",
          "padding": "15px",
          "right": "auto",
          "top": "40%",
          "transform": "translate(-50%, -50%)",
        },
      }
    }
  >
    <DialogBox__DialogBoxView
      className="ui-dialogbox"
    >
      <DialogBox__DialogBoxIconView>
        <DialogBox__StyledIcon
          iconName="times-circle"
          sizing="xxlarge"
          style={
            Object {
              "color": "#d9534f",
            }
          }
        />
      </DialogBox__DialogBoxIconView>
      <DialogBox__DialogBoxMessageContainerView>
        <DialogBox__DialogBoxMessageView
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>Sample error message</p>",
            }
          }
        />
        <DialogBox__ButtonBar>
          <DialogBox__StyledButtonText
            justify={2}
            noicon={true}
            onClick={[Function]}
            sizing="normal"
            text="Yes"
          />
          <DialogBox__Spacer />
          <DialogBox__StyledButtonText
            justify={2}
            noicon={true}
            onClick={[Function]}
            sizing="normal"
            text="No"
          />
        </DialogBox__ButtonBar>
      </DialogBox__DialogBoxMessageContainerView>
    </DialogBox__DialogBoxView>
  </Modal>
</Wrapper>
`;

exports[`Test the creation of an info DialogBox 1`] = `
<Wrapper
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  dialogType="info"
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  iconName="bomb"
  id={null}
  left={null}
  location="none"
  message="Sample info message"
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="DialogBox"
  onClose={[Function]}
  onError={[Function]}
  onOpen={[Function]}
  onSelection={[Function]}
  padding={null}
  reset={false}
  ripple={false}
  selected={false}
  show={true}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
>
  <Modal
    ariaHideApp={false}
    bodyOpenClassName="ReactModal__Body--open"
    closeTimeoutMS={0}
    contentLabel="DialogBox"
    isOpen={true}
    onAfterOpen={[Function]}
    onRequestClose={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={false}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {
          "borderRadius": "0",
          "bottom": "auto",
          "left": "50%",
          "marginRight": "-50%",
          "padding": "15px",
          "right": "auto",
          "top": "40%",
          "transform": "translate(-50%, -50%)",
        },
      }
    }
  >
    <DialogBox__DialogBoxView
      className="ui-dialogbox"
    >
      <DialogBox__DialogBoxIconView>
        <DialogBox__StyledIcon
          iconName="info-circle"
          sizing="xxlarge"
          style={
            Object {
              "color": "#5bc0de",
            }
          }
        />
      </DialogBox__DialogBoxIconView>
      <DialogBox__DialogBoxMessageContainerView>
        <DialogBox__DialogBoxMessageView
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>Sample info message</p>",
            }
          }
        />
        <DialogBox__ButtonBar>
          <DialogBox__StyledButtonText
            justify={2}
            noicon={true}
            onClick={[Function]}
            sizing="normal"
            text="Yes"
          />
          <DialogBox__Spacer />
          <DialogBox__StyledButtonText
            justify={2}
            noicon={true}
            onClick={[Function]}
            sizing="normal"
            text="No"
          />
        </DialogBox__ButtonBar>
      </DialogBox__DialogBoxMessageContainerView>
    </DialogBox__DialogBoxView>
  </Modal>
</Wrapper>
`;

exports[`Test the creation of an success DialogBox 1`] = `
<Wrapper
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  dialogType="success"
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  iconName="bomb"
  id={null}
  left={null}
  location="none"
  message="Sample success message"
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="DialogBox"
  onClose={[Function]}
  onError={[Function]}
  onOpen={[Function]}
  onSelection={[Function]}
  padding={null}
  reset={false}
  ripple={false}
  selected={false}
  show={true}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
>
  <Modal
    ariaHideApp={false}
    bodyOpenClassName="ReactModal__Body--open"
    closeTimeoutMS={0}
    contentLabel="DialogBox"
    isOpen={true}
    onAfterOpen={[Function]}
    onRequestClose={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={false}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {
          "borderRadius": "0",
          "bottom": "auto",
          "left": "50%",
          "marginRight": "-50%",
          "padding": "15px",
          "right": "auto",
          "top": "40%",
          "transform": "translate(-50%, -50%)",
        },
      }
    }
  >
    <DialogBox__DialogBoxView
      className="ui-dialogbox"
    >
      <DialogBox__DialogBoxIconView>
        <DialogBox__StyledIcon
          iconName="check-circle"
          sizing="xxlarge"
          style={
            Object {
              "color": "#5cb85c",
            }
          }
        />
      </DialogBox__DialogBoxIconView>
      <DialogBox__DialogBoxMessageContainerView>
        <DialogBox__DialogBoxMessageView
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>Sample success message</p>",
            }
          }
        />
        <DialogBox__ButtonBar>
          <DialogBox__StyledButtonText
            justify={2}
            noicon={true}
            onClick={[Function]}
            sizing="normal"
            text="Yes"
          />
          <DialogBox__Spacer />
          <DialogBox__StyledButtonText
            justify={2}
            noicon={true}
            onClick={[Function]}
            sizing="normal"
            text="No"
          />
        </DialogBox__ButtonBar>
      </DialogBox__DialogBoxMessageContainerView>
    </DialogBox__DialogBoxView>
  </Modal>
</Wrapper>
`;

exports[`Test the creation of an warning DialogBox 1`] = `
<Wrapper
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  dialogType="warning"
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  iconName="bomb"
  id={null}
  left={null}
  location="none"
  message="Sample warning message"
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="DialogBox"
  onClose={[Function]}
  onError={[Function]}
  onOpen={[Function]}
  onSelection={[Function]}
  padding={null}
  reset={false}
  ripple={false}
  selected={false}
  show={true}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
>
  <Modal
    ariaHideApp={false}
    bodyOpenClassName="ReactModal__Body--open"
    closeTimeoutMS={0}
    contentLabel="DialogBox"
    isOpen={true}
    onAfterOpen={[Function]}
    onRequestClose={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={false}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {
          "borderRadius": "0",
          "bottom": "auto",
          "left": "50%",
          "marginRight": "-50%",
          "padding": "15px",
          "right": "auto",
          "top": "40%",
          "transform": "translate(-50%, -50%)",
        },
      }
    }
  >
    <DialogBox__DialogBoxView
      className="ui-dialogbox"
    >
      <DialogBox__DialogBoxIconView>
        <DialogBox__StyledIcon
          iconName="exclamation-circle"
          sizing="xxlarge"
          style={
            Object {
              "color": "#f0ad4e",
            }
          }
        />
      </DialogBox__DialogBoxIconView>
      <DialogBox__DialogBoxMessageContainerView>
        <DialogBox__DialogBoxMessageView
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>Sample warning message</p>",
            }
          }
        />
        <DialogBox__ButtonBar>
          <DialogBox__StyledButtonText
            justify={2}
            noicon={true}
            onClick={[Function]}
            sizing="normal"
            text="Yes"
          />
          <DialogBox__Spacer />
          <DialogBox__StyledButtonText
            justify={2}
            noicon={true}
            onClick={[Function]}
            sizing="normal"
            text="No"
          />
        </DialogBox__ButtonBar>
      </DialogBox__DialogBoxMessageContainerView>
    </DialogBox__DialogBoxView>
  </Modal>
</Wrapper>
`;
